backend/de.metas.business/src/main/java/de/metas/costing/CostAmount.java

```diff
@@ -176,10 +183,38 @@ else if (percent.isOneHundred())
 		}
 	}
 
-	public CostAmount add(@NonNull final CostAmount amtToAdd)
+	public CostAmount addWithCurrencyConversionContext(@NonNull final CostAmount amtToAdd, @Nullable final CurrencyConversionContext conversionContext)
```

Comment:
my 2cents:
* why not simply call this method "add" (same like the other one)
* IMHO the algo shall be:  if the conversionContext is not null and currencies differ then convert the amtToAdd to the target currency; in both cases forward the converted amount to the other "add" method

PS: i would also evaluate if the conversionContext shall be `@NonNull` ... or if is for convenience, i am fine.